src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1987,7 +1987,7 @@ def _get_principal_dics(cli_ctx, role_assignments):
 
             graph_client = graph_client_factory(cli_ctx)
             principals = _get_object_stubs(graph_client, principal_ids)
-            return {i.object_id: (_get_displayable_name(i), i.object_type) for i in principals}
+            return {i['id']: (_get_displayable_name(i), i['@odata.type']) for i in principals}
```

Comment:
`@odata.type` is not the same as previous `object_type`:

https://github.com/Azure/azure-cli/blob/710c821b2920df504ae037e7b05763193b051a30/src/azure-cli/azure/cli/command_modules/role/custom.py#L39-L44

No conversion needed?

Reply:
Thanks for reminding!